Projects/Core/Services/SnapshotClient/SnapshotClient.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  SnapshotClient.swift
+//  Services
+//
+//  Created by GREEN on 6/5/24.
+//  Copyright © 2024 com.mashup.gabbangzip. All rights reserved.
+//
+
+import ComposableArchitecture
+import SwiftUI
+
+// MARK: - API Client Interface
+@DependencyClient
+public struct SnapshotClient {
+  var takeSnapshot: @Sendable () async throws -> UIImage
+}
+
+public extension DependencyValues {
+  var snapshotClient: SnapshotClient {
+    self[SnapshotClient.self]
+  }
+}
```

Comment:
P3:
snapshotClient에 다른 값(예: mock)을 사용할 수 있도록 set을 추가하는 것은 어떨까요?
cc. [Registering dependencies](https://pointfreeco.github.io/swift-dependencies/main/documentation/dependencies/registeringdependencies)
```suggestion
public extension DependencyValues {
  var snapshotClient: SnapshotClient {
    get { self[SnapshotClient.self] }
    set { self[SnapshotClient.self] = newValue }
  }
}
```

Reply:
아 좋습니다!